src/client.rs

```diff
@@ -214,30 +212,31 @@ pub type ApiResult<T> = Result<T, ApiError>;
 ///     .expect_success()
 ///     .expect_json();
 /// ```
-pub struct ApiResponse(ApiResult<reqwest::Response>);
+pub struct ApiResponse(ApiResult<reqwest::blocking::Response>);
 impl ApiResponse {
-    pub fn from(r: ApiResult<reqwest::Response>) -> ApiResponse {
+    pub fn from(r: ApiResult<reqwest::blocking::Response>) -> ApiResponse {
         ApiResponse(r)
     }
 
-    pub fn get(self) -> ApiResult<reqwest::Response> {
+    pub fn get(self) -> ApiResult<reqwest::blocking::Response> {
         self.0
     }
 
     pub fn expect_success(self) -> ApiResponse {
-        ApiResponse(self.0.and_then(move |mut response| {
-            if response.status().is_success() {
+        ApiResponse(self.0.and_then(move |response| {
+            let status = response.status();
+            if status.is_success() {
                 Ok(response)
             } else {
-                ApiErrorMessage::from_body(&mut response).and_then(|response_msg| {
-                    Err(ApiError::NonSuccess(response.status(), response_msg))
+                ApiErrorMessage::from_body(response).and_then(|response_msg| {
+                    Err(ApiError::NonSuccess(status, response_msg))
                 })
             }
         }))
```

Comment:
I'm not 100% confident that I changed this correctly

Reply:
Seems perfectly fine to me